### DISCUSSION TITLE
PRG not loading in Tang nano 20k with Shield.

### DISCUSSION THREAD

**Discussion by fhersete** [vossstef/tang_nano_20k_c64#77] [General]
Hello, I just installed the wonderful c64 core for the tang nano 20k with the till harbaum midi shield, the problem comes when loading the .PRG files. The CRTs work correctly, the D64s too, the TAPs also work, but when I select a prg on the osd, it starts up as if it were booting with a black screen, and then returns to the basic screen of the c64. The core I am using is the latest version that was published 5 days ago, the bl616 firmware is also the latest, along with its configuration, everything works correctly, except the PRGs, the kernal roms, I have tried with cbmdos, dolphin dos and jiffydos , and in none of them does it load the PRGs. Can someone tell me what is wrong with the loading of those PRGs?

**Comment by vossstef** [comment]
@fhersete the fpga bitsteam and fw i stored temporary directly in the branch i didn't tested thorougly yet due to lack of time. May i propose to use the files unter 'release' and try with thar. both fpga and fw need to be flashed. might be needed to delete the c64.ini

**Reply by fhersete** [reply]
The bitstream and firmware are updated, I have tried 1.2.3 and 1.3 from your github, and it is still the same, without being able to load the PRGs, it is strange, I have tried everything. The core works fine, it loads everything except the prgs. Greetings.

**Reply by fhersete** [reply]
I'm going to try releases of older versions, it's the only thing I need to try

**Reply by vossstef** [reply]
the files under release 1.2.3 should work and no complain yet. i tested a dual sid Demo as PRG and that was loading fine.

**Reply by fhersete** [reply]
Well, I don't understand it, I've tried several kernel roms, I've updated 3 versions of bitstream with their respective versions of bl616 and ini, and it's still the same, I'm using a till midi shield, I don't understand anything. jajajajaja.

**Reply by fhersete** [reply]
What file system do you use for the micro SD?. fat16, fat32, exfat?.

**Reply by fhersete** [reply]
I have tried all 3 file systems and nothing. What is the prg loading procedure, let's see if I'm doing it wrong or missing something, but come on, I could swear I've tried everything.

**Reply by vossstef** [reply]
i am using a fat32 sdcard. Are different PRG not loading?
 As mentioned i tried https://csdb.dk/release/?id=135891 and that is loading correctly.

**Reply by fhersete** [reply]
![IMG_20240702_131215](https://github.com/vossstef/tang_nano_20k_c64/assets/43078940/1680f773-a187-4de4-9ac0-cedfdfc1aa61)

I have tried with another tang 20k from scratch, without midi addon, with dupont cables, I have recorded the latest versions of firmware and roms and nothing, still the same, it does not load the prgs

**Comment by fhersete** [comment]
I have flashed the bitstream and the roms again with the gowin ide, and nothing, the prgs don't work, I don't understand it. I have tried all the file systems and the same thing.

**Reply by fhersete** [reply]
I don't know if the cause is the c64 midi addon, use the addon or connect the m0s via dupont directly to the tang nano 20k?

**Reply by fhersete** [reply]
nothing, there is no way to start the prg with the midi addon, I have tried everything that can be tried, I won't try it anymore.

**Reply by vossstef** [reply]
is it that PRG are loading correctly when MIDI is set to disabled? I will find some time over the nett weeks to Look into that.

**Reply by fhersete** [reply]
No, the prg does not load, neither with midi activated nor with it disabled, I have tried with addon and without addon and it remains the same. On the micro SD I have created a folder for each extension, crt, d64, prg, tap. I have put the boot in root of each one, c64prg.prg, c64crt.crt, and so on, the kernal in root and nothing., I press F12, I select storage, prg, I select the prg, I get a black screen for half a second , and it goes back to the basic of the c64, it does nothing.

**Reply by fhersete** [reply]
Video Test PRG - https://youtu.be/pr1v49AaXDI?si=LJAaLhZASyc06dYR

**Reply by vossstef** [reply]
thx. i will look next week into that. 

I think that i better block somehow PRG and TAP active at the same time as memory is cleared if you expect a PRG load and accidentally TAP isn't set to 'no disc' +save setting+ reset. Assume that an ongoing TAP load will spoil the PRG. 

Meaning: 
a) select a proper D64 or the default mountpoint. 
b) set all other loaders to 'no Disc' ! (bin,prg,crt,tap,flt) + save setting+ reset.
c) select desired PRG via OSD loader.
LIST
Check that there is at least 'something' in memory....
RUN
d) ???

**Reply by fhersete** [reply]
Okay, now it works, I have already identified the problem, in the OSD, if the tape sound is activated on on, the prgs are no longer loaded, it was that function. Now they load all the prgs, I disabled the loading of d64, crt and tap, but it still didn't load, and the tape sound on option was the one that was bothering me.

Thanks for everything Stefan, you have ported a wonderful core, it works very well, the plus4 could be ported. :)
